manager/apis/app/v1alpha1/datacatalog_types.go

```diff
@@ -5,11 +5,25 @@ package v1alpha1
 
 import "fybrik.io/fybrik/pkg/serde"
 
+type DataFlow string
+
+const (
+	// ReadFlow indicates a data set is being read
+	ReadFlow DataFlow = "read"
+
+	// WriteFlow indicates a data set is being written
+	WriteFlow DataFlow = "write"
+
+	// DeleteFlow indicates a data set is being deleted
+	DeleteFlow DataFlow = "delete"
+)
+
 // DataStore contains the details for accesing the data that are sent by catalog connectors
 // Credentials for accesing the data are stored in Vault, in the location represented by Vault property.
 type DataStore struct {
 	// Holds details for retrieving credentials by the modules from Vault store.
-	Vault Vault `json:"vault"`
+	// It is a map so that different credentials can be stored for the different DataStoreActions
+	Vault map[string]Vault `json:"vault"`
```

Comment:
Not sure I understand. It is a single data store, why different credentials?

Reply:
I think it used for different operations - read/write etc

Reply:
Yes, exactly.  @roee88 asked that we support different credentials for read/write/delete